From eac3b717859b18b6b61de7db8be3fb091287588b Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:02 +0800
Subject: [PATCH 030/150] dt-bindings: mtd: Add Nand Flash Controller support
 for Phytium SoC

This patch documents the DT binding for the Phytium Nand Flash controller.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: Ic7106061365f53b82f38cd2ef0c6b747ffc2ca9f
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../devicetree/bindings/mtd/phytium,nfc.yaml  | 42 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 43 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/mtd/phytium,nfc.yaml

diff --git a/Documentation/devicetree/bindings/mtd/phytium,nfc.yaml b/Documentation/devicetree/bindings/mtd/phytium,nfc.yaml
new file mode 100644
index 000000000000..4d3bac96d584
--- /dev/null
+++ b/Documentation/devicetree/bindings/mtd/phytium,nfc.yaml
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/mtd/phytium,nfc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium Nand Flash controller
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+properties:
+  compatible:
+    const: phytium,nfc
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    maxItems: 1
+
+  nand-ecc-strength:
+    const: 8
+
+  nand-ecc-step-size:
+    const: 512
+
+allOf:
+  - $ref: "nand-controller.yaml#"
+
+required:
+  - compatible
+  - reg
+  - interrupts
+
+examples:
+  - |
+    nand0: nand@28002000 {
+      compatible = "phytium,nfc";
+      reg = <0x0 0x28002000 0x0 0x1000>;
+      interrupts = <GIC_SPI 74 IRQ_TYPE_LEVEL_HIGH>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 8a01987f7f44..07733445b776 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2534,6 +2534,7 @@ F:	Documentation/devicetree/bindings/gpio/phytium,gpio.yaml
 F:	Documentation/devicetree/bindings/i2c/phytium,i2c.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,mci.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,sdci.yaml
+F:	Documentation/devicetree/bindings/mtd/phytium,nfc.yaml
 F:	Documentation/devicetree/bindings/net/can/phytium,can.yaml
 F:	Documentation/devicetree/bindings/sound/phytium,hda.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,qspi-nor.yaml
-- 
2.47.0

